src/protocols/Pager/Pager.cpp

```diff
@@ -243,9 +251,29 @@ int16_t PagerClient::transmit(uint8_t* data, size_t len, uint32_t addr, uint8_t
 int16_t PagerClient::startReceive(uint32_t pin, uint32_t addr, uint32_t mask) {
   // save the variables
   readBitPin = pin;
-  filterAddr = addr;
+
+  delete[] filterAddresses;
+  filterAddresses = new uint32_t[1];
+  filterAddresses[0] = addr;
+  filterMask = mask;
+
+  return startReceiveInternal();
+}
+
+int16_t PagerClient::startReceive(uint32_t pin, uint32_t *addr, uint32_t mask, size_t numAddress) {
+  // save the variables
+  readBitPin = pin;
+
+  delete[] filterAddresses;
+  filterAddresses = new uint32_t[numAddress];
+  memcpy(filterAddresses, addr, numAddress);
```

Comment:
Yeah, I was thinking about that too. So do you mean to keep a variable for when there's a single address to service the original version of the method, and then keep a pointer for a list of addresses?

What's the use case for masks anyway? I'm new to POCSAG, so I had assumed it would always be sufficient to pass in a list of addresses to cover all use cases.

Reply:
> do you mean to keep a variable for when there's a single address to service the original version of the method, and then keep a pointer for a list of addresses?

That's probably the simples way to do that. It actually flows nicely into the other comment about checking NULL address list - if it's null, check the address variable. If not, check the address list.

Reply:
> What's the use case for masks anyway?

It's very useful for checking continuous address ranges. If you need to match 256 addresses at time, it would be rather cumbersome having to spell them out in a list. Even better example is if you want to match all adresses - just set the mask to zero, no need to provide 2^20 addresses to check. However, this works best for continuous address ranges, that's why your PR is useful.

Reply:
That makes sense. If I were to set the mask to receive all messages, is there any risk that some would be dropped with high traffic or that is not a concern at all?

Reply:
> is there any risk that some would be dropped with high traffic or that is not a concern at all?

I don't think it will be an issue, though I did not test the implementation in a high traffic environment. Definitely not going to be na issue because of the address matching - if you set it to match everything, it will simply just alwasy return true whne checking the address.